worker/diskformatter/diskformatter.go

```diff
@@ -93,26 +99,48 @@ func (f *diskFormatter) Handle(diskNames []string) error {
 			)
 			continue
 		}
-		if blockDevices[i].FilesystemType != "" {
-			logger.Debugf("block device %q already has a filesystem", blockDevices[i].Name)
-			continue
-		}
 		storageInstance := result.Result
 		if storageInstance.Kind != storage.StorageKindFilesystem {
 			logger.Debugf("storage instance %q does not need a filesystem", storageInstance.Id)
 			continue
 		}
+		if storageInstance.Location != "" {
+			// TODO(axw) we need to ensure the mount point
+			// is still there, and remount if needed.
+			logger.Debugf("storage instance %q is already mounted at %q", storageInstance.Id, storageInstance.Location)
+			continue
+		}
 		devicePath, err := storage.BlockDevicePath(blockDevices[i])
 		if err != nil {
 			logger.Errorf("cannot get path for block device %q: %v", blockDevices[i].Name, err)
 			continue
 		}
-		if err := createFilesystem(devicePath); err != nil {
-			logger.Errorf("failed to create filesystem on block device %q: %v", blockDevices[i].Name, err)
+		if blockDevices[i].FilesystemType == "" {
+			if err := createFilesystem(devicePath); err != nil {
```

Comment:
I can't see how we know to create a file system as opposed to leaving it as a raw block device